### PR TITLE
Async transaction and multipart response parsing without using requests_toolbelt

### PR DESCRIPTION
**Pull request by miazamrai**
This PR is extension of Sameera's PR for async transaction handling. I made it to parse the multipart form data and extracting the output using pyarrow package. Also, it does not use requests_toolbelt package.

### CODE REVIEW

**Review by bradlo — APPROVED** [review]
This looks fine to me.

File: railib/api.py [hunk 1]
```diff
@@ -160,9 +158,26 @@ def _get_resource(ctx: Context, path: str, key=None, **kwargs) -> dict:
...
+    rsp = json.loads(rsp.read())
     return rsp[key] if key else rsp
 
+#
+# Finds and returns the multipart form-data boundary
+# Boundary is something like,
+# Content-Type: multipart/form-data; boundary=d6d47cadd395db7f6648a2ffb65751eb
+#
+
+
```
Comment (geokollias) [inline railib/api.py:170]: Drop empty lines?

**Review by geokollias — COMMENTED** [review]
Looks good. Thanks! Added few minor comments.

File: railib/api.py (hunk 1 above)
Comment (miazamrai) [reply railib/api.py:170]: I guess it is a convention ... my IDE was showing me lint issues if I was not adding extra two lines

File: requirements.txt [hunk 2]
```diff
@@ -1 +1,3 @@
 ed25519==1.5
+pyarrow==6.0.1
+setuptools==58.2.0
```
[the remark this answers is not in the document]
Comment (miazamrai) [reply requirements.txt:3]: Removed the setuptools

File: railib/api.py (hunk 1 above)
Comment (geokollias) [reply railib/api.py:170]: I think it's a PEP convention for function signatures/bodies. Weird to apply to doc strings.

Comment (geokollias) [reply railib/api.py:170]: I think if you remove the first empty `# ` line it'll stop showing that lint issue ;)

**Pull request merged**